0.7 release post (#146)
flowforge/website: src/blog/2022/07/flowforge-07-released.md

---
title: FlowForge 0.7 released
subtitle: Rollbacks Device Environment Variables and a FlowForge Theme
description: Rollbacks Device Environment Variables and a FlowForge Theme
date: 2022-07-07 12:00:00.0
authors: ["sam-machin"]
---
Rollback projects to a previous snapshot, Improvements in using Devices and more.

<!--more-->

> [TODO] Update video url and time below.

Reviewer: Agreed on the video placement. We should remove the generated image then.

Author: removing the image for a single post isn't a simple task, we can replace the image but the sizing is fixed so even a 1x1.png leaves a large blank space.
I propose we go with it as-is for this release and before the next one we create a custom template for release posts that sets the embedded YT video in the image space.

Author: discussed in https://flowforgeworkspace.slack.com/archives/C02248RCPJ4/p1657096621156179

Reviewer 2: Have raised https://github.com/flowforge/website/pull/148 which adds support for embedding a video in place of the hero image.

Reviewer 2: If/when #148 gets merged, I'll follow-up with the necessary tweaks to this post to make use of it.

Scroll down to read the details of whats in this release our you can watch our 1min roundup video of the 0.7 release

<iframe width="560" height="315" src="https://www.youtube.com/embed/BBNzFInc_2E" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

We're pleased to announce version 0.7 is now available. the next release of the FlowForge application.

## Features
[Rollback](https://github.com/flowforge/flowforge/issues/587)

> FlowForge is about running Node-RED at scale, part of that scale is having multiple users collaborate on the same project. When you are collaborating with people sometimes stuff needs to be undone. AS part of that we are introducing rollbacks, this means that you can now take a snapshot of your project at a point in time and then make changes safe in the knowledge that you can rollback to that previous snapshot if you need to.

Reviewer: ```suggestion
FlowForge is about running Node-RED at any scale, part of that scale is having multiple users collaborate on the same project. When you are collaborating with people it's important to be able to go back in time to a known working state. As part of that we are introducing rollbacks, this means that you can now take a snapshot of your project at a point in time and then make changes safe in the knowledge that you can rollback to that previous snapshot if you need to.
```

## Improvements
We've made a number of improvements to the overall experience of running FlowForge.

The Team Switch menu has been moved to a more prominent position in the interface, this also makes it easier to see how to create a new team.[https://github.com/flowforge/flowforge/issues/616]
If you are running your own copy of FlowForge you can now see the version details in the admin pages [https://github.com/flowforge/flowforge/issues/655]
Device polling is no longer an INFO level message  filling the log on your devices [https://github.com/flowforge/flowforge-device-agent/issues/10]

## Bug Fixes
[Devices now listen on all Interfaces allowing you to run local http servers](https://github.com/flowforge/flowforge-device-agent/issues/7)
[Fixed an error where a device gets an error  unknown device](https://github.com/flowforge/flowforge-device-agent/issues/7)
[The Audit Log in the Forge app displays the correct IP when a user logs in to Node-RED](https://github.com/flowforge/flowforge/issues/507)

> [Fixed an error with devices downloading snaphots from legacy stacks](https://github.com/flowforge/flowforge/issues/507)

Reviewer: ```suggestion
[Allow devices downloading snaphots from legacy stacks](https://github.com/flowforge/flowforge/issues/507)
```

Looking to change up the "Fixed ..." line over and over again, maybe this works?

[Fixed an error where objects in the node-red log would hang the log page](https://github.com/flowforge/flowforge/issues/735)
[Fixed an error where the Next Billing Date was incorrect](https://github.com/flowforge/flowforge/issues/745)
### Upgrading FlowForge

> [FlowForge Cloud](https://app.flowforge.com) is already running 0.7 - 

Reviewer: Should the double new line be removed so it's one paragraph? Ending a sentence/paragraph with `-` seems odd.

If you installed a previous version of FlowForge and want to upgrade, our documentation provides a
guide for [upgrading FlowForge on a local server](http://flowforge.com/docs/install#upgrade).

### Getting help

If you hit any problems with the platform, or have questions to ask, please do

> raise an [issue on GitHub](https://github.com/flowforge/flowforge/issues).

Reviewer: Do you want to call out `support@` for customers?

That also includes if you have any feedback or feature requests.

> We also have a `#flowforge` channel on the [Node-RED Slack workspace](https://nodered.org/slack).

Reviewer: 👍

---
title: FlowForge 0.7 released
subtitle: Rollbacks, Device Environment Variables and a FlowForge Theme
description: Rollbacks, Device Environment Variables and a FlowForge Theme
date: 2022-07-07 12:00:00.0
authors: ["sam-machin"]
video: BBNzFInc_2E
---

> Rollback projects to a previous snapshot, Improvements in using Devices and more.

Author: added

## Features
[Rollback](https://github.com/flowforge/flowforge/issues/587)
FlowForge is about running Node-RED at any scale, part of that scale is having multiple users collaborate on the same project. When you are collaborating with people it's important to be able to go back in time to a known working state. As part of that we are introducing rollbacks, this means that you can now take a snapshot of your project at a point in time and then make changes safe in the knowledge that you can rollback to that previous snapshot if you need to.

[Device Environment Variables](https://github.com/flowforge/flowforge/issues/680)
In the last release we introduced the concept of devices, we're already learning from how these are used and one feature we've added in 0.7 is Device Environment Variables, you have been able to set Environment Variables at the project level for a while now but when deploying a snapshot to multiple devices you may want to override these values for each device, eg to set a site ID or something, this feature enables that.

[FlowForge Theme](https://github.com/flowforge/flowforge-nr-theme/)

> Now that we have a stronger visual identity in the Forge application we have continued that work through to the Node-RED editor. If you are running a Node-RED 3.0 stack you will see a different theme in the editor. It's still very much Node-RED but just has some subtle hints to tie it back to the FlowForge application. We will continue to iterate on this to further integrate the experience between FlowForge and Node-RED in both directions.

Author: corrected wording to say that you need to either create or upgrade a project to pickup the new theme

![](../images/ff-07-theme.png)

[ProjectTypes](https://github.com/flowforge/flowforge/issues/380)

> The introduction of ProjectTypes is largely an internal feature to support our FlowForge Cloud proposition by being able to have different price points for projects based on the capacity and features available. While Stacks were a part of this ProjectTypes allows for an aggregation of stacks so that users can change versions of Node-RED while remaining on the same pricing tier.

Reviewer: If it's just a pricing differentiator feature that should be removed IMO, and when we 'release' the new tier we blog about that. For users, either administrators or end-users this paragraph doesn't influence their life, right?

Reviewer 2: I think it is worth mentioning, but the current wording (with so much emphasis on it being a 'largely internal feature') does perhaps miss the value to end-users (admins).

> The introduction of ProjectTypes is a way to group Stacks together that share common characteristics - such as memory/cpu limits, or the availability of particular features. In platforms with billing enabled, such as our own FlowForge Cloud, the ProjectTypes can have different price points set on them. Within FlowForge Cloud, you'll see we've introduced the Small ProjectType - which applies to all existing projects on the platform.

Author: agree Nick focussing on the feature first and then how we're utilising it in ff cloud is a good approach to this kind of thing.

Reviewer: Makes sense @knolleary . Thanks

---
title: FlowForge 0.7 released

> subtitle: Rollbacks Device Environment Variables and a FlowForge Theme
> description: Rollbacks Device Environment Variables and a FlowForge Theme

Reviewer 2: ```suggestion
subtitle: Rollbacks, Device Environment Variables and a FlowForge Theme
description: Rollbacks, Device Environment Variables and a FlowForge Theme
```
